### PR TITLE
Removed all remaining binaries from thirdparty

### LINKED ISSUE #5485: Remove managed & unmanaged binaries from the tree
I'm talking about the files in [thirdparty](https://github.com/OpenRA/OpenRA/tree/bleed/thirdparty). They're nice for packaging but they have no place in a source tree.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/2392392-remove-managed-unmanaged-binaries-from-the-tree?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #6863: Bundled thirdparty
Please provide build instructions for all the 3rdparty libraries etc.

In Fedora, distribution of prebuilt assemblies is not allowed. All has to be based on available sources, also for legal reasons.
https://fedoraproject.org/wiki/Packaging:Mono?rd=Packaging/Mono#Distributing_Prebuilt_Assemblies

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/5668567-bundled-thirdparty?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #8073: OpenRA goes out of memory
Sorry #7696, it's probably you.
```
Red Alert Mod at Version {DEV_VERSION}
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.34209
Exception of type `System.OutOfMemoryException`: An exception of type 'System.OutOfMemoryException' was drawn. (Eine Ausnahme vom Typ "System.OutOfMemoryException" wurde ausgelöst.)
   at OpenRA.Graphics.Sheet.CreateBuffer() \OpenRA\OpenRA.Game\Graphics\Sheet.cs:Line 149.
   at OpenRA.Graphics.Sheet.GetData() \OpenRA\OpenRA.Game\Graphics\Sheet.cs:Line 30.
   at OpenRA.Graphics.SpriteFont.CreateGlyph(Pair`2 c) \OpenRA\OpenRA.Game\Graphics\SpriteFont.cs:Line 120.
   at OpenRA.Primitives.Cache`2.get_Item(T key) \OpenRA\OpenRA.Game\Primitives\Cache.cs:Line 39.
   at OpenRA.Graphics.SpriteFont.PrecacheColor(Color c, String name) \OpenRA\OpenRA.Game\Graphics\SpriteFont.cs:Line 54.
   at OpenRA.Graphics.SpriteFont..ctor(String name, Int32 size, SheetBuilder builder) \OpenRA\OpenRA.Game\Graphics\SpriteFont.cs:Line 46.
   at OpenRA.Graphics.Renderer.<InitializeFonts>b__1(KeyValuePair`2 x) \OpenRA\OpenRA.Game\Graphics\Renderer.cs:Line 104.
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector)
   at OpenRA.Graphics.Renderer.InitializeFonts(Manifest m) \OpenRA\OpenRA.Game\Graphics\Renderer.cs:Line 103.
   at OpenRA.Game.InitializeMod(String mod, Arguments args) \OpenRA\OpenRA.Game\Game.cs:Line 307.
   at OpenRA.Game.Initialize(Arguments args) \OpenRA\OpenRA.Game\Game.cs:Line 264.
   at OpenRA.Program.Run(String[] args) \OpenRA\OpenRA.Game\Support\Program.cs:Line 109.
   at OpenRA.Program.Main(String[] args) \OpenRA\OpenRA.Game\Support\Program.cs:Line 39.
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/12717541-openra-goes-out-of-memory?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/6863 and
Closes https://github.com/OpenRA/OpenRA/issues/5485 together with https://github.com/OpenRA/OpenRA/pull/7933.
Fixes https://github.com/OpenRA/OpenRA/issues/8073.

These remaining CLI images were neither on NuGet nor have shipped releases somewhere else so I tagged releases from revisions that we picked. We also need to contribute some of our changes upstream like https://github.com/cdhowie/Eluant/pull/4.

### CODE REVIEW

File: thirdparty/fetch-thirdparty-deps.sh [hunk 1]
```diff
@@ -72,3 +72,13 @@ if [ ! -f FuzzyLogicLibrary.dll ]; then
 	cp ./FuzzyLogicLibrary.1.2.0/bin/Release/FuzzyLogicLibrary.dll .
 	rm -rf FuzzyLogicLibrary.1.2.0
 fi
+
+if [ ! -f SDL2-CS.dll ]; then
+	echo "Fetching SDL2 C# from GitHub."
+	wget --quiet https://github.com/OpenRA/SDL2-CS/releases/download/20140407/SDL2-CS.dll
```
Comment (pchote) [inline thirdparty/fetch-thirdparty-deps.sh]: wget is not installed by default under OSX.  Could you please use curl?

Comment (Mailaender) [inline thirdparty/fetch-thirdparty-deps.sh]: Sure.

File: packaging/package-all.sh [hunk 2]
```diff
@@ -22,9 +22,9 @@ make package
 find . -path "*.mdb" -delete
 
 test -e Changelog.md && rm Changelog.md
-wget https://raw.githubusercontent.com/wiki/OpenRA/OpenRA/Changelog.md
+curl -s -L -o Changelog.md https://raw.githubusercontent.com/wiki/OpenRA/OpenRA/Changelog.md
 
-wget http://daringfireball.net/projects/downloads/Markdown_1.0.1.zip
+curl -s -L -o Markdown_1.0.1.zip https://raw.githubusercontent.com/wiki/OpenRA/OpenRA/Changelog.md
```
Comment (pchote) [inline packaging/package-all.sh]: Copypasta url error.

Comment (Mailaender) [inline packaging/package-all.sh]: Fixed.

File: thirdparty/fetch-thirdparty-deps.sh [hunk 3]
```diff
@@ -72,3 +72,13 @@ if [ ! -f FuzzyLogicLibrary.dll ]; then
 	cp ./FuzzyLogicLibrary/bin/Release/FuzzyLogicLibrary.dll .
 	rm -rf FuzzyLogicLibrary
 fi
+
+if [ ! -f SDL2-CS.dll ]; then
+	echo "Fetching SDL2 C# from GitHub."
```
Comment (pchote) [inline thirdparty/fetch-thirdparty-deps.sh]: SDL2-CS

**Comment by pchote** [comment]
Can't get this to work.  What am I missing?
```
paul@MacBook-Air:OpenRA+pr/7953$ git clean -fX
[...]
paul@MacBook-Air:OpenRA+pr/7953$ ./configure
[...]
paul@MacBook-Air:OpenRA+pr/7953$ make dependencies
[...]
paul@MacBook-Air:OpenRA+pr/7953$ make
[...]
paul@MacBook-Air:OpenRA+pr/7953$ mono --debug OpenRA.Game.exe
Platform is OSX
Using SDL 2 with OpenGL renderer
Renderer initialization failed. Fallback in place. Check graphics.log for details.
Using SDL 2 with OpenGL renderer
Renderer initialization failed. Fallback in place. Check graphics.log for details.
Exception of type `System.InvalidOperationException`: No suitable renderers were found. Check graphics.log for details.
  at OpenRA.Game.Initialize (OpenRA.Arguments args) [0x0031a] in /Users/paul/src/OpenRA/OpenRA.Game//Game.cs:268 
  at OpenRA.Program.Run (System.String[] args) [0x00007] in /Users/paul/src/OpenRA/OpenRA.Game//Support/Program.cs:109 
  at OpenRA.Program.Main (System.String[] args) [0x00050] in /Users/paul/src/OpenRA/OpenRA.Game//Support/Program.cs:39 
paul@MacBook-Air:OpenRA+pr/7953$ cat ~/Library/Application\ Support/OpenRA/Logs/graphics.log
System.DllNotFoundException: libSDL2.dylib
  at (wrapper managed-to-native) SDL2.SDL:SDL_Init (uint)
  at OpenRA.Renderer.Sdl2.Sdl2GraphicsDevice..ctor (Size windowSize, WindowMode windowMode) [0x00013] in /Users/paul/src/OpenRA/OpenRA.Renderer.Sdl2//Sdl2GraphicsDevice.cs:46 
  at OpenRA.Renderer.Sdl2.DeviceFactory.Create (Size size, WindowMode windowMode) [0x0000d] in /Users/paul/src/OpenRA/OpenRA.Renderer.Sdl2//Sdl2GraphicsDevice.cs:29 
  at OpenRA.Graphics.Renderer.CreateDevice (System.Reflection.Assembly rendererDll, Int32 width, Int32 height, WindowMode window) [0x0004a] in /Users/paul/src/OpenRA/OpenRA.Game//Graphics/Renderer.cs:90 
  at OpenRA.Graphics.Renderer..ctor (OpenRA.GraphicSettings graphicSettings, OpenRA.ServerSettings serverSettings) [0x00082] in /Users/paul/src/OpenRA/OpenRA.Game//Graphics/Renderer.cs:55 
  at OpenRA.Game.Initialize (OpenRA.Arguments args) [0x00196] in /Users/paul/src/OpenRA/OpenRA.Game//Game.cs:238 
System.DllNotFoundException: libSDL2.dylib
  at (wrapper managed-to-native) SDL2.SDL:SDL_Init (uint)
  at OpenRA.Renderer.Sdl2.Sdl2GraphicsDevice..ctor (Size windowSize, WindowMode windowMode) [0x00013] in /Users/paul/src/OpenRA/OpenRA.Renderer.Sdl2//Sdl2GraphicsDevice.cs:46 
  at OpenRA.Renderer.Sdl2.DeviceFactory.Create (Size size, WindowMode windowMode) [0x0000d] in /Users/paul/src/OpenRA/OpenRA.Renderer.Sdl2//Sdl2GraphicsDevice.cs:29 
  at OpenRA.Graphics.Renderer.CreateDevice (System.Reflection.Assembly rendererDll, Int32 width, Int32 height, WindowMode window) [0x0004a] in /Users/paul/src/OpenRA/OpenRA.Game//Graphics/Renderer.cs:90 
  at OpenRA.Graphics.Renderer..ctor (OpenRA.GraphicSettings graphicSettings, OpenRA.ServerSettings serverSettings) [0x00082] in /Users/paul/src/OpenRA/OpenRA.Game//Graphics/Renderer.cs:55 
  at OpenRA.Game.Initialize (OpenRA.Arguments args) [0x00196] in /Users/paul/src/OpenRA/OpenRA.Game//Game.cs:238 
```

File: thirdparty/fetch-thirdparty-deps.sh (hunk 3 above)
Comment (Mailaender) [inline thirdparty/fetch-thirdparty-deps.sh]: https://github.com/OpenRA/SDL2-CS/blob/master/README#L1

**Comment by Mailaender** [comment]
Rebased. Try `make clean && make all`.

**Comment by obrakmann** [comment]
Works for me on Linux, :+1:

**Comment by Phrohdoh** [comment]
Fixed it for me under OS X 10.3.3. :+1:

**Comment by RoosterDragon** [comment]
At the end of multiple 'make clean'.

```
Time Elapsed 00:00:00.18
rmdir : Cannot find path
'C:\Users\Roost\Documents\Actual\Programming\C#\OpenRA\thirdparty\windows\'
because it does not exist.
At C:\Users\Roost\Documents\Actual\Programming\C#\OpenRA\make.ps1:77 char:3
+         rmdir thirdparty/windows/
+         ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Roost\...dparty\window
   s\:String) [Remove-Item], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.Remov
   eItemCommand

Clean complete.
Press enter to continue.
```

Seems harmless but it does show up as a big red error message which would be nice to avoid.

Otherwise: Builds & runs on Windows.

File: packaging/osx/buildpackage.sh [hunk 4]
```diff
@@ -14,7 +14,7 @@ if [ -e "OpenRA.app" ]; then
     exit 2
 fi
 
-wget https://github.com/OpenRA/OpenRALauncherOSX/releases/download/${LAUNCHER_TAG}/launcher.zip || exit 3
+curl -s -L -o launcher.zip https://github.com/OpenRA/OpenRALauncherOSX/releases/download/${LAUNCHER_TAG}/launcher.zip || exit 3
```
Comment (pchote) [inline packaging/osx/buildpackage.sh]: All these filenames are the same as the remote name, so you should use the more concise `-O` everwhere.

Comment (Mailaender) [inline packaging/osx/buildpackage.sh]: Why didn't you come up with that last week. Now this is buried by lot's of other commits.

Comment (Phrohdoh) [inline packaging/osx/buildpackage.sh]: `git rebase` is simple enough.

Comment (Mailaender) [inline packaging/osx/buildpackage.sh]: My client broke with the last update sadly. https://github.com/git-cola/git-cola/issues/445

**Comment by Mailaender** [comment]
Updated.

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/e6d50ca557933be58f9acbea194587d11bdccd8d%5E...e6d50ca557933be58f9acbea194587d11bdccd8d

**Comment by Mailaender** [comment]
Small [addition](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e6d50ca557933be58f9acbea194587d11bdccd8d...3423d2132d8dd48e74b78e4fd274230bde38b497)

**Comment by pchote** [comment]
The downside of all these changes is that it is now tricky to do development while on a plane or in another area without internet access unless you have the foresight to stash a spare copy of all the files somewhere.

**Comment by Mailaender** [comment]
Once all the old branches are merged/rebased you won't need to `make clean` all the time when testing old stuff.

**Pull request merged**